File: generated_program.py
import requests


"""
Calls the OpenAI API to generate a response based on the provided prompt.
:param content: (optional, default: 'write a haiku about ai')
:return message: 
"""
def agent(content):
    url = 'http://localhost:3000/agent'
    data = {
        "content": content,
    }
    response = requests.post(url, json=data)
    return response.json()



"""
Determines whether a given number is Magical.
:param number: (required)
:return number: 
    :return isMagical: 
    :return message: 
"""
def isMagical(number):
    url = 'http://localhost:3000/isMagical'
    data = {
        "number": number,
    }
    response = requests.post(url, json=data)
    return response.json()




import random

def main():
    # Generate a poem with numbers only
    numbers = [random.randint(0, 100) for _ in range(10)]  # 10 random numbers
    poem = ' '.join(map(str, numbers))  # Create a poem by joining numbers as strings

    print("Generated Poem (numbers only):")
    print(poem)

    # Check for magical numbers
    magical_numbers = []
    for number in numbers:
        result = isMagical(number)
        if result.get('isMagical'):
            magical_numbers.append(number)

    print("Magical Numbers:")
    print(magical_numbers)

if __name__ == "__main__":
    main()

File: generated_program.js
const axios = require('axios');


/**
* Calls the OpenAI API to generate a response based on the provided prompt.
 * @param {string} content - (optional, default: 'write a haiku about ai')
 * @returns {Object} - The result object.
 *   @property {string} message - 
*/
async function agent(content) {
    const url = 'http://localhost:3000/agent';
    const data = {
        content: content,
    };
    const response = await axios.post(url, data);
    return response.data;
}



/**
* Determines whether a given number is Magical.
 * @param {number} number - (required)
 * @returns {Object} - The result object.
 *   @property {number} number - 
 *   @property {boolean} isMagical - 
 *   @property {string} message - 
*/
async function isMagical(number) {
    const url = 'http://localhost:3000/isMagical';
    const data = {
        number: number,
    };
    const response = await axios.post(url, data);
    return response.data;
}




async function main() {
    // Create a poem using numbers
    const poem = `
    1 2 3 4 5
    6 7 8 9 10
    11 12 13 14 15
    16 17 18 19 20
    `;

    // Extract numbers from the poem
    const numbers = poem.match(/\d+/g).map(Number);
    
    // Check which numbers are magical
    const magicalResults = await Promise.all(numbers.map(num => isMagical(num)));

    // Filter and display magical numbers
    const magicalNumbers = magicalResults.filter(result => result.isMagical).map(result => result.number);
    
    console.log("The following numbers are magical:", magicalNumbers);
}

main();
